src/invidious/helpers/utils.cr

```diff
@@ -471,3 +471,81 @@ def fetch_random_instance
 
   return filtered_instance_list.sample(1)[0]
 end
+
+struct LanguageEntry
+  class InvalidLanguageEntry < Exception
+  end
+
+  property tag, subtags, quality : Float64
+
+  def initialize(@tag : String, @subtags : Array(String), @quality : Float64)
+    @quality = @quality.clamp(0.0, 1.0)
+  end
+
+  def to_s(io)
+    io << tag
+    unless subtags.empty?
+      io << '-' << @subtags.join '-'
+    end
+  end
+
+  def self.from_string(language_str : String) : self
+    range_and_quality = language_str.split ';'
```

Comment:
`range` here refers to a locale right? As in `fr-CH` or `en-GB`? Imo calling it `range` is rather confusing; it should be renamed.

Reply:
Agreed. I was just following the spec language but it's rather confusing.